frontend/server/handlers/artifacts.ts

```diff
@@ -239,6 +254,91 @@ function getGCSArtifactHandler(options: { key: string; bucket: string }, peek: n
   };
 }
 
+function getVolumeArtifactsHandler(options: { bucket: string; key: string }, peek: number = 0) {
+  const { key, bucket } = options;
+  return async (req: Request, res: Response) => {
+    try {
+      const [pod, err] = await serverInfo.getHostPod();
+      if (err) {
+        res.status(500).send(err);
+        return;
+      }
+
+      if (!pod) {
+        res.status(500).send('Could not get server pod');
+        return;
+      }
+
+      // volumes not specified or volume named ${bucket} not specified
+      if (!Array.isArray(pod?.spec?.volumes) || !pod.spec.volumes.find(v => v?.name === bucket)) {
+        const message = `Failed to open volume://${bucket}/${key}, volume ${bucket} not found`;
+        res.status(404).send(message);
+        return;
+      }
+
+      let serverContainer: V1Container | undefined;
+      if (Array.isArray(pod?.spec?.containers)) {
+        serverContainer = pod?.spec.containers.find(c => c.name === 'ml-pipeline-ui');
+      }
+
+      if (!serverContainer) {
+        const message = `Failed to open volume://${bucket}/${key}, server deployment container ml-pipeline-ui not found`;
+        res.status(404).send(message);
+        return;
+      }
+
+      if (!Array.isArray(serverContainer.volumeMounts)) {
+        const message = `Failed to open volume://${bucket}/${key}, volume mount not found`;
+        res.status(404).send(message);
+        return;
+      }
+
+      // find volumes mount
+      const volumeMount = serverContainer.volumeMounts.find(v => {
+        // volume name must be same
+        if (v?.name !== bucket) {
+          return false;
+        }
+        // if volume subPath set, volume subPath must be prefix of key
+        if (v?.subPath) {
+          return key.startsWith(v.subPath);
+        }
+        return true;
+      });
+
+      // volumes mount named ${bucket} not found
+      if (!volumeMount) {
+        const message = `Failed to open volume://${bucket}/${key}, volume mount ${bucket} not found`;
+        res.status(404).send(message);
+        return;
+      }
+
+      // finally file path
+      const filePath = pruneAndConcatPath(volumeMount.mountPath, key, volumeMount.subPath);
+
+      if (!fs.existsSync(filePath)) {
```

Comment:
nit: using fs sync methods in a request handler is really a bad idea, it blocks all the other requests.

Can you change this to a promise based call https://nodejs.org/dist/latest-v10.x/docs/api/fs.html#fs_fs_promises_api, so that you can use an `async function` to handle promises?